extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-module.ts

```diff
@@ -99,13 +109,38 @@ export class DataExtensionsEditorModule {
               return;
             }
 
+            if (!isQueryLanguage(db.language)) {
```

Comment:
Could we combine this with the check at https://github.com/github/vscode-codeql/blob/1289ab509c5f55c7757628f904fea68acca71e80/extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-module.ts#L63 so we aren't checking the language twice and using different error messages.

So the two checks we're performing are:
- Is the language a member of the `QueryLanguage` type?
- Is the language supported by the data extensions editor?

Maybe something like:
```typescript
if (!(SUPPORTED_LANGUAGES.includes(language) && isQueryLanguage(language))) {
  void showAndLogErrorMessage(
    this.app.logger,
    `The data extensions editor is not supported for ${db.language} databases.`,
  );
  return;
}
```

I'd put both of these checks at the start where we current check `SUPPORTED_LANGUAGES`.

Reply:
Discussed sync and resolved!